### PR TITLE
Python 2/3 compatibility

### PR DESCRIPTION
**Pull request by daf**
These changes make the codebase run on both Python 2.6+ and Python 3.x (3.5 tested) without using a shim layer like `six`.

Most changes are simple: `print` -> `print()`, `has_key` -> `in`, `xrange()` -> `range()`, and `iteritems()` -> `items()`, some list comprehensions.

Only tricky bit is the urllib import pattern for 2/3 compatibility.

### CODE REVIEW

**Comment by ChrisBarker-NOAA** [comment]
Thanks Dave -- we do want to go to py3 at some point, so this is good. And thanks for not using six -- I'm not a fan ;-)

We'll take a look and see if we see any issues.

File: libgoods/libgoods/noaa_coops.py [hunk 1]
```diff
@@ -1,6 +1,12 @@
 #!/usr/bin/env python
+from __future__ import print_function
 import datetime
-import urllib2, requests
```
Comment (ChrisBarker-NOAA) [inline libgoods/libgoods/noaa_coops.py]: we should probably use the requests package anyway -- but this should be fine.

Comment (daf) [inline libgoods/libgoods/noaa_coops.py]: I agree!

**Comment by ChrisBarker-NOAA** [comment]
This all looks good to me. The only issue is that we'll need to get used to typing the parentheses with print...

But this code is @AmyMacFadyen's baby -- so I'll let her make a final decision and merge.

**Pull request merged**